corehq/apps/couch_sql_migration/couchsqlmigration.py

```diff
@@ -742,6 +776,22 @@ def get_queue_obj_id(queue_obj):
             return getattr(queue_obj, queue_id_param)
         self.get_queue_obj_id = get_queue_obj_id
 
+        def get_cached_list_key():
+            return "partial_queues.queued_or_processing.%s" % run_timestamp
+        self.get_cached_list_key = get_cached_list_key
+
+    def add_processing_doc_id(self, doc_id):
+        client = get_redis_connection()
+        client.rpush(self.get_cached_list_key(), doc_id)
+
+    def remove_processing_doc_id(self, doc_id):
+        client = get_redis_connection()
+        client.lrem(self.get_cached_list_key(), 1, doc_id)
```

Comment:
I would be surprised if it got to be too long i was operating under the assumption that if it were longer than [10000](https://github.com/dimagi/commcare-hq/blob/efcfd074e1ec404c7654f45e2166462e96b374bb/corehq/apps/couch_sql_migration/couchsqlmigration.py#L127), we should [slow down growth](https://github.com/dimagi/commcare-hq/blob/efcfd074e1ec404c7654f45e2166462e96b374bb/corehq/apps/couch_sql_migration/couchsqlmigration.py#L168) and try to process as much from it as possible. i'd be surprised if it ever got there tho. 

i could spin it off into a greenlet since this could be a bottleneck and, since redis is single threaded before v4 (which i don't think we're on), and is supposed to be thread safe, it wouldn't be an issue

Reply:
19aec5e